bundles/framework/layerlist/components/LayerCollapse.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { List } from '../../../admin/admin-layereditor/components/List';
+import { Layer } from './LayerCollapse/Layer';
+import { Collapse, Panel } from '../../../admin/admin-layereditor/components/Collapse';
+import { Badge } from '../../../admin/admin-layereditor/components/Badge';
+import { Alert } from '../../../admin/admin-layereditor/components/Alert';
+
+const getLayers = (layers, otherProps) => layers.map(
+    (lyr, index) => {
+        const props = {
+            model: lyr,
+            even: index % 2 === 0,
+            ...otherProps
+        };
+        return <Layer key={index} {...props}/>;
+    }
+);
+const getPanels = (groups, searchResults, otherProps) => {
+    const panels = (searchResults || groups).map((cur, index) => {
+        const group = searchResults ? cur.group : cur;
+        const filteredLayers = searchResults ? cur.layers : null;
+
+        let badgeText = group.getLayers().length;
+        if (filteredLayers && filteredLayers.length !== group.getLayers().length) {
+            badgeText = filteredLayers.length + ' / ' + badgeText;
+        }
+        const badgeProps = {
+            inversed: true,
+            count: badgeText
+        };
+        const layers = filteredLayers || group.getLayers();
+        const listProps = {
+            dataSource: getLayers(layers, otherProps),
+            bordered: false
+        };
+        const panelProps = {
+            header: group.getTitle(),
+            extra: <Badge {...badgeProps}/>,
+            children: <List {...listProps}/>
+        };
+        return <Panel key={index} {...panelProps}/>;
+    });
+    return panels;
+};
+
+const getEmptyMessage = filterKeyword => {
+    const alertProps = {
+        message: 'No layers found',
+        description: 'TODO: Show search keyword used: ' + filterKeyword,
+        type: 'info',
+        showIcon: true
+    };
+    return <Alert {...alertProps}/>;
+};
+
+const getSearchResults = (groups, keyword) => {
+    if (!keyword && keyword !== 0) {
+        return null;
+    }
+    const results = groups.map(group => {
+        const layers = group.getLayers()
+            .filter(lyr => group.matchesKeyword(lyr.getId(), keyword)); // and some other rule?
+        return {group, layers};
+    }).filter(result => result.layers.length !== 0);
+    return results;
+};
+
+export const LayerCollapse = props => {
+    const { groups, filterKeyword, ...rest } = props;
```

Comment:
Maybe destructure already in function signature?
`export const LayerCollapse = ({ groups, filterKeyword, ...rest }) => {...}`

Reply:
Destructing in function signature done